### ISSUE TITLE
wizard should warn when trying to configure portable mode if installed to "c:\program files" & friends

### ISSUE THREAD

**Issue by wchristian** [kvirc/KVIrc#2194]
Portable mode causes it to try and store settings and such in the install dir, however for non-admin applications that dir is unwritable normally:

![image](https://cloud.githubusercontent.com/assets/175467/23104457/942ebe4e-f6ce-11e6-85fd-4dce88343433.png)

**Comment by wodim** [comment]
That's because you installed normally and then installed the portable verison, no?

**Comment by wchristian** [comment]
This happened on my very first install.

I don't remember the flow exactly, but at some point in the installer there was a checkbox for portable, so i clicked it. I think the path was already chosen at that point.

**Comment by wodim** [comment]
The problem with portable is that you're not supposed to install the application to a folder where you'll be unable to write to later as a normal user. Portable means (and the setup program tells you, I think) that settings will be stored alongside the application itself.

**Comment by wchristian** [comment]
Right, and I'm saying the installer should warn about that, or as i wrote
in the title, outright deny it.

**Comment by wchristian** [comment]
Oops, remembered a bit of additional info. It is not the installer that offers portable mode, but the wizard on kvirc first run.

So i guess that one should do a double check that all locations for settings files are writable and deny portable with an explanation and instruction to reinstall.

**Comment by ltc-data** [comment]
> Oops, remembered a bit of additional info. It is not the installer that offers portable mode, but the wizard on kvirc first run.

wchristian This is the problem exactly you nailed it.
I had a look at this and NSIS could be reworked to include the portable install option to message user that you need to run the portable mode in the setup wizard too. (for first time install)

Also I found that when you upgrade the existing portable install NSIS doesnt remember your portable path and as far as I know NSIS doesnt have any option to do this except some ugly hack perhaps like writing a registry key for path thats not removed on upgrade and removed if unistalling program only.

**Comment by IceN9ne** [comment]
The whole point of portable mode is to avoid the registry and keep everything needed to run the program in one place, so that it can be portable between computers or similar use cases.

If the user doesn't understand what portable mode is and how it works, then it isn't KVIrc's fault for allowing them to install inside of a location that requires elevated rights to write to.  If you choose to initially install in the default application directory, then you don't want  portable mode.

The only change the should be made is to disable the portable mode checkbox in the first-start wizard if the application install directory doesn't allow writing from the current user.

**Comment by wchristian** [comment]
> If you choose to initially install in the default application directory, then you don't want portable mode.

I have in fact several programs installed that handle this fine, so i disagree with some (but not all) of your thinking there. That said:

> The only change the should be made is to disable the portable mode checkbox in the first-start wizard if the application install directory doesn't allow writing from the current user.

That's fine too. I don't care whether it's a warning, or just plain disables the box. Though as a thought, it might be weird if the box disables without explanation, so maybe a text in a popup or something would be good.
